core/domain/suggestion_services.py

```diff
@@ -1826,3 +2020,866 @@ def update_question_suggestion(
+
+    Returns:
+        QuestionReviewStats. The corresponding QuestionReviewStats domain
+        object.
+    """
+    return suggestion_registry.QuestionReviewStats(
+        question_review_stats_model.reviewer_user_id,
+        question_review_stats_model.topic_id,
+        question_review_stats_model.reviewed_questions_count,
+        question_review_stats_model.accepted_questions_count,
+        (
+            question_review_stats_model
+            .accepted_questions_with_reviewer_edits_count),
+        question_review_stats_model.first_contribution_date,
+        question_review_stats_model.last_contribution_date
+    )
+
+
+def _get_all_translation_review_stats(
+    user_id: str
+) -> List[suggestion_registry.TranslationReviewStats]:
+    """Gets all TranslationReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(TranslationReviewStats). TranslationReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    translation_review_stats_models = (
+        suggestion_models.TranslationReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_translation_review_stats_from_model(model)
+        for model in translation_review_stats_models
+    ]
+
+
+def _get_all_question_contribution_stats(
+    user_id: str
+) -> List[suggestion_registry.QuestionContributionStats]:
+    """Gets all QuestionContributionStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionContributionStats). QuestionContributionStats domain
+        objects corresponding to the supplied user.
+    """
+    question_contribution_stats_models = (
+        suggestion_models.QuestionContributionStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_contribution_stats_from_model(model)
+        for model in question_contribution_stats_models
+    ]
+
+
+def _get_all_question_review_stats(
+    user_id: str
+) -> List[suggestion_registry.QuestionReviewStats]:
+    """Gets all QuestionReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionReviewStats). QuestionReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    question_review_stats_models = (
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_model(model)
+        for model in question_review_stats_models
+    ]
+
+
+# TODO(https://shorturl.at/deLSV): pre-fetching and caching of stats data
```

Comment:
All TODOs should point to an issue on our issue tracker. See other TODOs in the codebase for examples.

Reply:
Done